core/src/contracts/stablex_contract.rs

```diff
@@ -350,6 +359,21 @@ impl StableXContract for StableXContractImpl {
         }
         .boxed()
     }
+
+    fn get_nonce<'a>(&'a self) -> BoxFuture<'a, Result<U256>> {
+        use futures::compat::Future01CompatExt as _;
+        async move {
+            let web3 = self.instance.raw_instance().web3();
+            let account = self.account().ok_or_else(|| anyhow!("no account"))?;
+            let address = account.address();
+            web3.eth()
+                .transaction_count(address, None)
```

Comment:
I think ultimately we might even want to share this state across run-loops so that the next batch uses the current batch's nonce +1 even if the cancellation tx is still pending (right now it would still use the same nonce since transaction_count is based on only mined txs).

Reply:
Made an issue for this.